src/spec.c

```diff
@@ -1275,6 +1275,8 @@ void IndexSpec_FreeInternals(IndexSpec *spec) {
   } else {
     thpool_add_work(cleanPool, (thpool_proc)IndexSpec_FreeUnlinkedData, spec);
   }
+
+  pthread_rwlock_destroy(&spec->rwlock);
```

Comment:
How or where do we validate that the lock is not locked?

Reply:
see https://stackoverflow.com/questions/55614415/pthread-rwlock-destroy-of-a-locked-mutex
the plan is to have the spec ref counted. will explain in person